eng/pipelines/libraries/base-job.yml

```diff
@@ -0,0 +1,105 @@
+  archType: ''
+  osSubgroup: ''
+  framework: ''
+  isOfficialBuild: false
+  timeoutInMinutes: 150
+  container: ''
+  steps: []
+  dependsOn: []
+  variables: {}
+  name: ''
+  displayName: ''
+  testScope: ''
+  pool: ''
+  stepName: ''
+
+jobs:
+  - template: /eng/common/templates/job/job.yml
+    parameters:
+      name: ${{ format('libraries_{0}_{1}_{2}{3}_{4}_{5}', parameters.stepName, parameters.framework, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
+      ${{ if eq(parameters.framework, 'netcoreapp') }}:
+        displayName: ${{ format('{0} {1}{2} {3} {4}', parameters.stepName, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
+      ${{ if ne(parameters.framework, 'netcoreapp') }}:
+        displayName: ${{ format('{0} {1} {2} {3} {4}', parameters.stepName, parameters.osGroup, parameters.framework, parameters.archType, parameters.buildConfig) }}
+
+      enableTelemetry: ${{ parameters.isOfficialBuild }} # TODO: figure out if it's needed
+      container: ${{ parameters.container }}
+      helixRepo: dotnet/runtime
+      pool: ${{ parameters.pool }}
+      variables:
+        - _BuildConfig: ${{ parameters.buildConfig }}
+        - _msbuildCommonParameters: ''
+        - _stripSymbolsArg: ''
+        - _runtimeOSArg: ''
+        - _portableArg: ''
+        - _finalFrameworkArg: -framework ${{ parameters.framework }}
+        - _buildScript: $(buildScriptFileName)$(scriptExt)
+        - _warnAsErrorArg: ''
+        - _testScopeArg: ''
+
+        - ${{ if ne(parameters.testScope, '') }}:
+          - _testScopeArg: -testscope ${{ parameters.testScope }}
+
+        - ${{ if and(eq(parameters.osGroup, 'Linux'), eq(parameters.osSubGroup, '_rhel6')) }}:
+          - _runtimeOSArg: /p:RuntimeOS=rhel.6
+          - _portableArg: /p:PortableBuild=false
+        
+        - ${{ if and(eq(parameters.osGroup, 'Linux'), eq(parameters.osSubGroup, '_musl')) }}:
+          - _runtimeOSArg: /p:RuntimeOS=linux-musl
+        
+        - ${{ if and(eq(parameters.osGroup, 'Linux'), eq(parameters.osSubGroup, ''), eq(parameters.archType, 'arm')) }}:
+          - _runtimeOSArg: /p:RuntimeOS=ubuntu.16.04
+
+        - ${{ if and(or(eq(parameters.osGroup, 'Linux'), eq(parameters.osGroup, 'WebAssembly')), ne(parameters.container, '')) }}:
+          - _warnAsErrorArg: '-warnAsError false'
+
+        - ${{ if eq(parameters.osGroup, 'WebAssembly') }}:
+          - _runtimeOSArg: -os ${{ parameters.osGroup }}
+
+        - ${{ if eq(parameters.framework, 'allConfigurations' ) }}:
+          - _finalFrameworkArg: -allConfigurations
+
+        - ${{ if eq(parameters.isOfficialBuild, 'true') }}:
+          - _msbuildCommonParameters: /p:OfficialBuildId=$(Build.BuildNumber)
+
+        # Windows variables
+        - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+          - _msbuildCommand: powershell -ExecutionPolicy ByPass -NoProfile eng\common\msbuild.ps1 -warnaserror:0 -ci
```

Comment:
why do we need warnaserror false here?

Reply:
Actually this was ported to xplat-setup and I can remove it. And I also think it was added for a reason I can’t remember anymore. So I will remove it in a follow up PR.

Reply:
Thanks.

> And I also think it was added for a reason I can’t remember anymore.

That's exactly was referring to above with "lack of context" :)